spec/support/fake_analytics.rb

```diff
@@ -73,6 +75,10 @@ def track_event(event, original_attributes = {})
   module UndocumentedParamsChecker
     mattr_accessor :asts
     mattr_accessor :docstrings
+    DOCUMENTATION_OPTIONAL_PARAMS = [
+      :user_id,
+      *AnalyticsEventsDocumenter::DOCUMENTATION_OPTIONAL_PARAMS.map(&:to_sym),
+    ].uniq.freeze
```

Comment:
A/N: There is a mismatch between how UndocumentedParamsChecker and AnalyticsEventsDocumenter load method definitions for events. 

It is beyond the scope of this PR to fully fix these behavioral discrepancies, so this is a workaround until a better solution is found.